library/ui/src/main/java/com/google/android/exoplayer2/ui/SubtitleView.java

```diff
@@ -386,7 +386,12 @@ private Cue removeEmbeddedStyling(Cue cue) {
           cue.buildUpon().setTextSize(Cue.DIMEN_UNSET, Cue.TYPE_UNSET).clearWindowColor();
       if (cueText != null) {
         // Remove all spans, regardless of type.
-        strippedCue.setText(cueText.toString());
+        strippedCue.setText(new SpannableString(cueText.toString()));
```

Comment:
Since we're no longer stripping all Spans, I think it's probably easier just to directly copy the `Spanned` to a `SpannableString` (including **all** the spans) by passing it into the constructor, and then iterate over them calling `removeSpan()` unless the type is recognised as a 'language feature'.

I suspect it will also reduce the number of `instanceof` and cast calls needed (especially if you store it in a `SpannableString` local before calling `Cue#setText` later).

It seems less complicated than being sure we definitely copied all the flags etc over.